src/mmw/js/src/core/views.js

```diff
@@ -745,11 +755,22 @@ var MapView = Marionette.ItemView.extend({
         this.fitToAoi();
     },
 
-    fitToAoi: function() {
-        var areaOfInterest = this.model.get('areaOfInterest');
+    /**
+    Fits the current map view to an AOI if one is set.
+    @param layer -- A leaflet layer of the AOI if one is available.
+                    If not provided, a new L.GeoJSON layer is created to
+                    obtain the bounds of the AOI.
+    **/
+    fitToAoi: function(layer) {
+        if(!layer) {
+            var areaOfInterest = this.model.get('areaOfInterest');
+
+            if (areaOfInterest) {
+                layer = new L.GeoJSON(areaOfInterest);
+            }
+        }
 
-        if (areaOfInterest) {
-            var layer = new L.GeoJSON(areaOfInterest);
+        if(layer) {
```

Comment:
Does this `if` condition need to be here? Seems like `layer` will always be defined by this time.

Reply:
If the layer is not passed in, but there is also no areaOfInterest defined, then layer here will be undefined.

Reply:
Ah I see now, thanks for pointing that out. 👍